packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualRows.ts

```diff
@@ -154,6 +155,12 @@ export const useGridVirtualRows = (
       if (containerProps.isVirtualized && page !== nextPage) {
         setRenderingState({ virtualPage: nextPage });
         logger.debug(`Changing page from ${page} to ${nextPage}`);
+
+        apiRef.current.publishEvent(GRID_VIRTUAL_PAGE_CHANGE, {
+          currentPage: page,
+          nextPage,
+          api: apiRef,
+        });
```

Comment:
This seems to be an implementation detail of the virtualization. I don't think that we should expose it.

```suggestion
```

Instead, prefer listening to the viewport

Reply:
I can call the API here directly but that would deviate from the established pattern. I added that event because of it. We can not document the event and keep it private?

Reply:
Could we listen to the scroll event instead?

Reply:
I can listen to any of the scroll events we currently have but I would still need to calculate where the last currently virtualized row is and when the new one comes in.
I know about the current limitations of the virtualization, we discussed them in the last meeting with @dtassone and @m4theushw (the `scrollToIndex` jump problem). The way I see it even if the logic of the virtualization changes would still be a notion of what is currently not virtualized so a potential refactoring would not be much. That is the only part that potentially wound need to change.